File: php/src/GetToken.php
<?php

namespace HyperSwitchAI\Examples;

require __DIR__ . '/../vendor/autoload.php';

use Dotenv\Dotenv;
use GuzzleHttp\Client;
use GuzzleHttp\Exception\GuzzleException;

// Load environment variables from .env file
$dotenv = Dotenv::createImmutable(__DIR__ . '/..');
$dotenv->load();

const BASE_URL = 'https://api.hyperswitchai.com';

try {
    // Check for required environment variables
    if (!isset($_ENV['USERNAME']) || !isset($_ENV['PASSWORD'])) {
        throw new \Exception('USERNAME and PASSWORD must be set in a .env file in the php directory. You can rename .env.sample to .env and fill in your own email and password.');
    }

    // Create HTTP client
    $client = new Client();

    // Make authentication request
    $response = $client->post(BASE_URL . '/auth', [
        'headers' => [
            'Content-Type' => 'application/json'
        ],
        'json' => [
            'username' => $_ENV['USERNAME'],
            'password' => $_ENV['PASSWORD']
        ]
    ]);

    // Parse response
    $data = json_decode($response->getBody(), true);
    $token = $data['token'];

    // Output success message
    echo "\n\033[32m✅ Token successfully retrieved\033[0m\n";
    echo "Token: " . $token . "\n";

} catch (GuzzleException $e) {
    echo "Error: Authentication failed - " . $e->getMessage() . "\n";
} catch (\Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
} 

File: python/src/get_token.py
import os
import json
from dotenv import load_dotenv
import requests

# Load environment variables from .env file
load_dotenv()

BASE_URL = 'https://api.hyperswitchai.com'

def main():
    try:
        # Check for required environment variables
        if not os.getenv('USERNAME') or not os.getenv('PASSWORD'):
            raise ValueError('USERNAME and PASSWORD must be set in a .env file in the python directory. '
                           'You can rename .env.sample to .env and fill in your own email and password.')
        
        # Make authentication request
        auth_response = requests.post(
            f'{BASE_URL}/auth',
            headers={'Content-Type': 'application/json'},
            json={
                'username': os.getenv('USERNAME'),
                'password': os.getenv('PASSWORD')
            }
        )
        
        # Check if request was successful
        auth_response.raise_for_status()
        
        # Extract token from response
        token = auth_response.json()['token']
        
        # Print success message with color
        print('\n\033[92m✅ Token successfully retrieved\033[0m')
        print('Token:', token)

    except requests.exceptions.RequestException as e:
        print('Error: Authentication failed')
        print('Details:', str(e))
    except Exception as e:
        print('Error:', str(e))

if __name__ == '__main__':
    main()
